### PR TITLE
Add libs folder to dll resolve path

### PR DESCRIPTION
**Pull request by d-uzlov**
Closes https://github.com/mlcommons/mobile_app_open/issues/232

### CODE REVIEW

**Comment by freedomtan** [comment]
@shabanim please help review this.

**Comment by shabanim** [comment]
I made some comments on the code changes. I need to discuss this with Danil before approving it.

**Review by mkankana — APPROVED** [review]

**Review by shabanim — COMMENTED** [review]
Good!

File: flutter/lib/backend/list.dart [hunk 1]
```diff
@@ -28,7 +28,7 @@ class BackendInfo {
         continue;
       }
       if (Platform.isWindows) {
-        path = './libs/$path';
+        path = '$path.dll';
```
[the remark this answers is not in the document]
Comment (d-uzlov) [reply flutter/lib/backend/list.dart:31]: TFlite backend consists of one file, so it didn't have any issues even before this PR. We specified the full relative path `./libs/...`, so the app was able to load it.
This exact line doesn't fix anything, but it proves that changes in this PR work as intended.
`$path.dll` is still located in the `libs` folder, so if we are able to load it without specifying the path, then we should also be able to load all the transitive dependencies from that folder.

File: flutter/cpp/backends/external.cc [hunk 2]
```diff
@@ -63,37 +63,84 @@ std::string BackendFunctions::isSupported(const std::string& lib_path,
...
+  const auto setDllResult = SetDllDirectoryW(libFolder.c_str());
+  if (setDllResult == 0) {
+    const auto errorCode = GetLastError();
+    LOG(ERROR) << "Unable to add './libs' as a library path (error  "
+               << errorCode
+               << "): " << std::system_category().message(errorCode);
+    return nullptr;
+  }
+
+  auto handle = tflite::SharedLibrary::LoadLibrary(wide_lib_path.c_str());
```
[the remark this answers is not in the document]
Comment (d-uzlov) [reply flutter/cpp/backends/external.cc:118]: We always use TFlite utility functions everywhere.
We use it for loading, decompressing and transforming images, for logging, for loading dynamic libraries.
We don't load it separately, it is statically linked as a part of the backend_bridge.dll, which is the essential part of the application.

We only link part of the TFlite that is actually needed for this stuff.
If you disable tflite backend, most of the tflite won't even be compiled.

**Comment by d-uzlov** [comment]
@shabanim, please, use "Squash and merge" option for merging pull requests when possible.
It keeps commit history linear.
When using conventional merge commits branch history becomes very messy very fast, especially if you merge something old. Github still keeps link to the pull request so we don't really loose anything when squashing.

**Pull request merged**